packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -127,23 +132,81 @@ Future<void> checkGradleDependencies() async {
   progress.stop();
 }
 
+/// Tries to migrate [settings.gradle] by removing the code that load plugins as
+/// Gradle subprojects. If this file was modified by the developer, then prints
+/// an error message indicating how to migrate the file manually.
+void _removePluginsFromSettingsGradle(Directory androidDirectory) {
```

Comment:
I don't see a discussion of this breaking change in the document. If the automatic migration fails on a transitive dependency, then does the user end up stuck?

Reply:
what do you mean by "fails on transitive dependency"?  I changed it such that the current `settings.gradle` is renamed as `.settings.gradle`.

Reply:
This was under the section "Migration Plan".